src/controllers/CategoryFacetQueryController.ts

```diff
@@ -60,4 +49,27 @@ export class CategoryFacetQueryController {
       allowedValuesPatternType: AllowedValuesPatternType.Regex
     });
   }
+
+  private shouldAddFilterToQuery(path: string[]) {
+    return path.length != 0 && !Utils.arrayEqual(path, this.categoryFacet.options.basePath);
+  }
+
+  private addQueryFilter(queryBuilder: QueryBuilder, path: string[]) {
+    if (this.shouldAddFilterToQuery(path)) {
+      queryBuilder.advancedExpression.addFieldExpression(this.categoryFacet.options.field as string, '==', [
+        path.join(this.categoryFacet.options.delimitingCharacter)
+      ]);
+    }
+  }
+
+  private addCategoryFacetRequest(queryBuilder: QueryBuilder, path: string[], maximumNumberOfValues: number) {
+    const categoryFacetsRequest: ICategoryFacetRequest = {
+      field: this.categoryFacet.options.field as string,
```

Comment:
Curious about the `as string` cast here and on line 59. What is the usual type?

Reply:
`IField`